lib/brewdler/dsl.rb

```diff
@@ -1,23 +1,47 @@
 module Brewdler
   class Dsl
+    class Entry
+      attr_reader :type, :name, :extra
+
+      def initialize(type, name, extra={})
+        @type = type
+        @name = name
+        @extra = extra
+      end
+    end
+
+    attr_reader :entries
+
     def initialize(input)
       @input = input
+      @entries = []
     end
 
     def process
       instance_eval(@input)
+      self
```

Comment:
Not too weird to me. It is used a lot in JAVA to do something like `object.foo().bar()`